presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/operators/PartitionAndSerialize.h"
+#include "presto_cpp/main/operators/ShuffleInterface.h"
+#include "presto_cpp/main/types/PrestoToVeloxQueryPlan.h"
+#include "velox/core/PlanNode.h"
+#include "velox/exec/HashPartitionFunction.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/functions/prestosql/aggregates/RegisterAggregateFunctions.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+#include "velox/parse/TypeResolver.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+#include <gtest/gtest.h>
+
+using namespace facebook::presto::operators;
+
+namespace facebook::velox::exec::test {
+
+namespace {
+class HashPartitionFunctionSpec : public core::PartitionFunctionSpec {
```

Comment:
We seem to duplicate this code in multiple places. Is this necessary?

Reply:
I plan to move the duplicate codes into planNode.h in Velox in followup. Then we can one place for all the partition function definitions.